### PR TITLE
circleci: use ubuntu 20.04

### PR DESCRIPTION
**Pull request by kolyshkin**
/kind dependency-change

#### What this PR does / why we need it:

Changes circle-ci base image from ubuntu 16.04 to ubuntu 20.04.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by kolyshkin** [comment]
OK it looks like some of the steps we take (disabling firewall, adding iptables rules, I guess) breaks ubuntu 20.04 networking, so `apt` stops working:

```
+ sudo apt update
...   
  Temporary failure resolving 'archive.ubuntu.com'
Err:2 http://ppa.launchpad.net/git-core/ppa/ubuntu focal InRelease             
  Temporary failure resolving 'ppa.launchpad.net'
```
etc

File: scripts/circle-setup [hunk 1]
```diff
@@ -33,7 +33,7 @@ prepare_system() {
     sudo sysctl -w net.ipv4.conf.all.route_localnet=1
     sudo sysctl -w net.bridge.bridge-nf-call-iptables=1
     sudo sysctl -w net.ipv4.ip_forward=1
-    sudo iptables -t nat -I POSTROUTING -s 127.0.0.1 ! -d 127.0.0.1 -j MASQUERADE
+    #   sudo iptables -t nat -I POSTROUTING -s 127.0.0.1 ! -d 127.0.0.1 -j MASQUERADE
```
Comment (TomSweeneyRedHat) [inline scripts/circle-setup:36]: For the final submit, can you remove these commented lines or add a comment as to why they're commented out?

Comment (kolyshkin) [reply scripts/circle-setup:36]: This is a temporary test commit, I think I marked it as such. Some of this is breaking networking, trying to figure out what as I can't repro this locally.

**Comment by kolyshkin** [comment]
OK, this is mostly working now, the only thing that is broken is the "ping pod from another pod" test case which was earlier fixed via commit 0628b3dc88e55ef7f by `sudo ufw disable`. It looks like disabling the firewall breaks Ubuntu 20.04 networking, so trying out alternatives.

**Comment by kolyshkin** [comment]
OK, so crictl can't be run since masquerade rule breaks DNS lookups, Similar issue: https://github.com/kubernetes-sigs/cri-tools/pull/626

**Comment by kolyshkin** [comment]
I made it working!

**Comment by kolyshkin** [comment]
critest is now passing: https://app.circleci.com/pipelines/github/cri-o/cri-o/6479/workflows/55331d2f-651a-4d99-8030-32cf5ffff5b6/jobs/104261

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**